libraries/helpers.rb

```diff
@@ -34,15 +34,29 @@ def ingredient_package_name
       product_lookup(new_resource.product_name, version_string(new_resource.version))['package-name']
     end
 
-    def install_mixlib_versioning
-      # We need Mixlib::Versioning in the library helpers for
-      # parsing the version string.
-      chef_gem "#{new_resource.product_name}-mixlib-versioning" do # ~FC009 foodcritic needs an update
-        package_name 'mixlib-versioning'
-        compile_time true
+    def ensure_mixlib_versioning_gem_installed!
+      node.run_state[:mixlib_versioning_gem_installed] ||= begin # ~FC001
+        install_gem_from_rubygems('mixlib-versioning', '1.1.0')
```

Comment:
I wonder if we should expose some attributes for the gem version (and optional git ref) like we do in artifactory-pro:
https://github.com/chef-cookbooks/artifactory-pro/blob/master/attributes/client.rb#L20-L51

This would allow you to rev the version of the gems in cookbooks that consume this library cookbook.